### ISSUE TITLE
Support for OpenShift 4.5

### ISSUE THREAD

**Issue by kincl** [NVIDIA/gpu-operator#82]
### 1. Issue or feature description

Per the Installation documentation[1] OpenShift 4.5 is not supported by the gpu-operator. What needs to happen to get OpenShift 4.5 supported? 

Per Red Hat's documented OpenShift Life Cycle Policy[2] a OpenShift minor release is moved from Full Support to Maintenance Support a month after the next minor release is out so 4.4 has been in Maintenance support since August 13. 

[1] https://docs.nvidia.com/datacenter/kubernetes/openshift-on-gpu-install-guide/index.html
[2] https://access.redhat.com/support/policy/updates/openshift

**Comment by dmc5179** [comment]
I believe this is a mismatch between the version of the kernel running on the RedHat CoreOS nodes and the version of the kernel inside the gpu driver container image. It's been awhile since I looked at this. Back in 4.3 I modified the gpu driver image to build it on a connected system for use in disconnected environments. At the time the gpu operator would yum install some packages when loaded.

**Comment by aheifetz** [comment]
DanC.  good to see you on-line.  We need support for OpenShift 4.5  also, why is the entitlement hard coded for RHEL and not other operating systems?

pkg/controller/clusterpolicy/object_controls.go

    if osTag != "rhel" {
		return nil
	}

	entitlementPath := "/etc/pki/entitlements"
	if _, err := os.Stat(entitlementPath); os.IsNotExist(err) {
		log.Info(fmt.Sprintf("ERROR: Could not find RedHat entitlement on current node at path %s", entitlementPath))
		os.Exit(1)
	}

**Comment by dmc5179** [comment]
@aheifetz I was able to get the GPU driver working in a 4.5.9 cluster this week using a nvidia driver container image that I created myself. Nvidia based their GPU operator on Red Hat's UBI images but installs RPMs that are not part of the UBI repos. It uses RPMs that are part of core RHEL. This is why a RHEL sub is required to build the driver images for the GPU operator when used on OpenShift 4. The easiest way around this is to just build the driver independently on an internet connected RHEL server and then update the helm chart to use your own images. I do this in ansible with helm like:

```
        /usr/local/bin/helm install \
            nvidia-gpu-operator \
            --namespace default \
            --set 'platform.openshift=true,operator.defaultRuntime=crio' \
            --set "driver.repository={{ mirror_registry }}:{{ mirror_registry_port }}/nvidia"
            --set "driver.version=450.51.06-1.0.0-custom-rhcos-4.18.0-193.13.2.el8_2.x86_64-4.5.2"
            --set "toolkit.repository={{ mirror_registry }}:{{ mirror_registry_port }}/nvidia"
            --set "devicePlugin.repository={{ mirror_registry }}:{{ mirror_registry_port }}/nvidia"
            --set "dcgmExporter.repository={{ mirror_registry }}:{{ mirror_registry_port }}/nvidia"
            '{{ mirror_base }}/helm/gpu-operator-1.1.7.tgz'
```

**Comment by geoberle** [comment]
@dmc5179 how did you build your own driver container image for a specific openshift 4.x version?

**Comment by shivamerla** [comment]
@geoberle new operator version v1.3.x published to RH catalog supports 4.4/4.5/4.6.
